generate_commitment/generate_submarine_commit.py

```diff
@@ -161,35 +179,149 @@ def generateAddressBexport(fromAddress, toAddress, sendAmount, dappData, gasPric
+        rlp.encode(tx))
+
+
+def _get_args():
+    '''
+    Internal function. Creates an argparser for the main method to use.
+
+    :return: parser: argparse object for parsing program arguments.
+    '''
+    parser = argparse.ArgumentParser(
+        description=
+        "Tool to create a TXUnlock transaction for submarine commitments",
+        formatter_class=argparse.RawTextHelpFormatter)
+    parser.add_argument(
+        '-t',
+        '--target-address',
+        required=True,
+        type=str,
+        default="",
+        help="Target end address to send the money to in the end. "
+        "Probably this should be the LibSubmarine contract address.")
+    parser.add_argument(
+        '-f',
+        '--from-address',
+        required=True,
+        type=str,
+        default="",
+        help=
+        "From address that starts the submarine process. This should be an address "
+        "that you control.")
+    parser.add_argument(
+        '-a',
+        '--amount',
+        required=True,
+        type=int,
+        default=0,
+        help=
+        "Amount of money you are sending through the submarine transaction in Wei."
+    )
+    parser.add_argument(
+        '-d',
+        '--dapp-data',
+        required=False,
+        type=str,
+        default="",
+        help=
+        "Optional DApp Data field. Pass any additional function parameters here "
+        "as needed. Data will be interpreted as a hex string, e.g. 0x414243... "
+        "In most cases this should be left blank.")
+    parser.add_argument(
+        '-p',
+        '--gas-price',
+        required=False,
+        type=int,
+        default=50000000000,
+        help="Optional Gas price for TX Unlock transaction. Default is 50 GWei"
+    )
+    parser.add_argument(
+        '-l',
+        '--gas-limit',
+        required=False,
+        type=int,
+        default=3712394,
+        help=
+        "Optional Gas limit for TX Unlock transaction. Default is 3.7 MWei.")
+    return parser.parse_args()
 
 
 def main():
+    '''
+    Main method. Runs the program if it is used standalone (rather than as an exported library).
+    '''
 
-    #init test variables
-    gasPrice = 50000000000  # 50 gwei
-    gasLimit = 3712394  # startgas
-    toAddress = normalize_address("0x7AEB1Fd3A42731c4Ae80870044C992eb689fb2Fe")
-    fromAddress = normalize_address("0x94146296881b3322838cFF5f2d50fdde841928D8")
-    sendAmount = 123000000000000
-    dappData = b""
+    parser = _get_args()
+
+    if len(parser.target_address) != 42:
+        log.error(
+            "Target Address length does not appear to match the correct length of an Ethereum address"
+        )
+        sys.exit(1)
+    if len(parser.from_address) != 42:
+        log.error(
+            "From Address length does not appear to match the correct length of an Ethereum address"
+        )
+        sys.exit(2)
+    if parser.target_address[0:2] != "0x":
```

Comment:
Might also be worth checking that the supplied addresses have a valid checksum encoding using `check_checksum(addr)`. 
So much ether has been lost due to simple copy-paste mistakes!

Reply:
addressed